pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/XPathPanelController.java

```diff
@@ -112,9 +120,63 @@ public void initialize(URL location, ResourceBundle resources) {
                            // Reevaluate XPath anytime the expression or the XPath version changes
                            .or(xpathVersionProperty().changes())
                            .subscribe(tick -> parent.refreshXPathResults());
+
+        xpathExpressionArea.plainTextChanges()
+                           .map(m -> m.getInserted())
+                           .filter(t -> t.matches("[a-zA-Z/]"))
+                           .subscribe(e -> {
+                               if (e.matches("/")) {
```

Comment:
Then you feed it to the autocomplete method, which should take both the input string and the position of the slash as parameters (they're both in the tuple). You'll use the position of the slash when you replace the text by the suggestion when a suggestion is selected